src/AspNet.Security.OAuth.BattleNet/BattleNetAuthenticationHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using AspNet.Security.OAuth.Extensions;
+using Microsoft.AspNet.Authentication;
+using Microsoft.AspNet.Authentication.OAuth;
+using Microsoft.AspNet.Http.Authentication;
+using Microsoft.Framework.Internal;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.BattleNet {
+    public class BattleNetAuthenticationHandler : OAuthAuthenticationHandler<BattleNetAuthenticationOptions> {
+        public BattleNetAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+
+			string userInfoUri = Options.UserInformationEndpoint + "?access_token=" + tokens.AccessToken;
+            var request = new HttpRequestMessage(HttpMethod.Get, userInfoUri);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+            var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            response.EnsureSuccessStatusCode();
+
+            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+
+			string id = BattleNetAuthenticationHelper.GetIdentifier(payload);
```

Comment:
Is there a reason that one would use var over explict types besides for line length or dynamic like objects? I get the remove the variable altogether, which ill do, but I usually do it because it is much more readable